include/seastar/core/metrics_api.hh

```diff
@@ -324,6 +324,7 @@ class impl {
     config _config;
     bool _dirty = true;
     shared_ptr<metric_metadata> _metadata;
+    std::set<sstring> _labels;
```

Comment:
Sadly, prometheus.md doesn't even mention the concept of labels, it probably should?
ScyllaDB's docs/dev/metrics.md has an introduction to labels, if you're looking for a source to copy.